src/main/webapp/github-scm-source.js

```diff
@@ -5,47 +5,23 @@ Behaviour.specify("input[name$=_configuredByUrlRadio]", 'GitHubSCMSourceRadioCon
         return;
     }
 
-    var getNthParent = function(e, n) {
-        while (n > 0) {
-            if (e.parentNode) {
-                e = e.parentNode;
-                n--;
-            } else {
-                return null;
-            }
-        }
-        return e;
-    };
-
-    // Todo: Replace with a query selector?
-    var findNeighboringDynamicInput = function(e) {
-        var inputTbody = getNthParent(e, 4 /*tbody > tr > td > label > input*/);
-        if (inputTbody) {
-            // input hidden is always in the 4th position
-            var hiddenBlock = inputTbody.childNodes[4].firstElementChild.firstElementChild;
-            return hiddenBlock
-        }
-    };
-
-    var neighboringDynamicInput = findNeighboringDynamicInput(e);
-    if (neighboringDynamicInput) {
+    var configuredByUrlInput = e.closest('div[name="source"]').querySelector('input[name="configuredByUrl"]');
+    if (configuredByUrlInput) {
         e.onclick = function() {
-            neighboringDynamicInput.value = e.value;
+            configuredByUrlInput.value = e.value;
             // When changing to true the event is triggered.
-            if (e.value == "false") {
+            if (e.value === "false") {
                 // When changing to false a trigger has to be fired in order to fetch the repos from the backend
                 if (document.createEvent) {
```

Comment:
Could you make these 2 into guard clauses? there is a lot of nesting in this function.

Reply:
can't see any reason to keep this anyway